hathor/consensus/poa/poa.py

```diff
@@ -48,13 +62,14 @@ def calculate_weight(settings: PoaSettings, block: PoaBlock, signer_index: int)
     return BLOCK_WEIGHT_IN_TURN if expected_index == signer_index else BLOCK_WEIGHT_OUT_OF_TURN
 
 
-def get_signer_index_and_public_key(settings: PoaSettings, signer_id: bytes) -> tuple[int, bytes] | None:
+def get_signer_index_and_public_key(settings: PoaSettings, block: PoaBlock) -> tuple[int, bytes] | None:
     """Given a `signer_id`, return its signer index and public key bytes. Return `None` if not found."""
     from hathor.consensus.poa import PoaSigner
-    sorted_signers = sorted(settings.signers)
+    active_signers = get_active_signers(settings, block.get_height())
+    sorted_signers = sorted(active_signers)
 
     for i, public_key_bytes in enumerate(sorted_signers):
```

Comment:
This code has the same regression as here: https://github.com/HathorNetwork/hathor-core/pull/1071#discussion_r1667075303

Reply:
Done

Reply:
Great! Thanks!